lightning-block-sync/src/rpc.rs

```diff
@@ -0,0 +1,182 @@
+use crate::{BlockHeaderData, BlockSource, AsyncBlockSourceResult};
+use crate::http::{HttpClient, HttpEndpoint, JsonResponse};
+
+use bitcoin::blockdata::block::Block;
+use bitcoin::hash_types::BlockHash;
+use bitcoin::hashes::hex::ToHex;
+
+use serde_json;
+
+use std::convert::TryFrom;
+use std::convert::TryInto;
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+/// A simple RPC client for calling methods using HTTP `POST`.
+pub struct RpcClient {
+	basic_auth: String,
+	endpoint: HttpEndpoint,
+	client: HttpClient,
+	id: AtomicUsize,
+}
+
+impl RpcClient {
+	/// Creates a new RPC client connected to the given endpoint with the provided credentials. The
+	/// credentials should be a base64 encoding of a user name and password joined by a colon, as is
+	/// required for HTTP basic access authentication.
+	pub fn new(credentials: &str, endpoint: HttpEndpoint) -> std::io::Result<Self> {
+		let client = HttpClient::connect(&endpoint)?;
+		Ok(Self {
+			basic_auth: "Basic ".to_string() + credentials,
+			endpoint,
+			client,
+			id: AtomicUsize::new(0),
```

Comment:
Not sure if conclusions were reached, but we discussed how the ID wasn't being fully utilized in the current PRs and could potentially be pushed off.